src/main/java/data/TaskManager.java

```diff
@@ -361,7 +362,25 @@ private static LocalDate findDateFromDayNumber(WeekView weekView, MonthView mont
             checkIfDateInCurrentMonth(date);
 
         } else {
-            date = weekView.getStartOfWeek().plusDays(dayInt - 1);
+            LocalDate startOfWeek = weekView.getStartOfWeek();
+            LocalDate endOfWeek = startOfWeek.plusDays(6);
+            LocalDate startOfMonth = startOfWeek.withDayOfMonth(1);
+            LocalDate possibleDate = startOfMonth.plusDays(dayInt - 1);
+            
+            boolean isBeforeStartOfWeek = possibleDate.isBefore(startOfWeek);
+            boolean isNotInSameMonth = possibleDate.getMonth() != startOfWeek.getMonth();
+            boolean dayIntRefersToNextMonth = isBeforeStartOfWeek || isNotInSameMonth;
+
+            if (dayIntRefersToNextMonth) {
+                LocalDate startOfNextMonth = startOfMonth.plusMonths(1).withDayOfMonth(dayInt);
+                if (startOfNextMonth.isAfter(endOfWeek)) {
+                    throw new TaskManagerException(NOT_CURRENT_WEEK_MESSAGE);
+                }
+                date = startOfNextMonth;
+            } else {
+                date = possibleDate;
+            }
+
```

Comment:
Great job using variables with descriptive names to make your code more readable and reuse long strings of commands in shorter variables